stable/admin/templates/_helpers.tpl

```diff
@@ -135,22 +135,30 @@ imagePullSecrets:
 Get Pod securityContext
 */}}
 {{- define "securityContext" -}}
-{{- if eq (include "defaultfalse" .Values.admin.securityContext.enabled) "true" }}
+{{- if or (eq (include "defaultfalse" .Values.admin.securityContext.enabled) "true") (eq (include "defaultfalse" .Values.admin.securityContext.runAsNonRootGroup) "true") (eq (include "defaultfalse" .Values.admin.securityContext.fsGroupOnly) "true") }}
 securityContext:
-  runAsUser: {{ default 1000 .Values.admin.securityContext.runAsUser }}
-  runAsGroup: 0
   fsGroup: {{ default 1000 .Values.admin.securityContext.fsGroup }}
-  {{- include "sc.fsGroupChangePolicy" . }}
+  {{- include "sc.fsGroupChangePolicy" . | indent 2 }}
+  {{- include "sc.runAs" . | indent 2 }}
+{{- end }}
+{{- end -}}
+
+{{/*
+Get security context runAsUser and runAsGroup
+*/}}
+{{- define "sc.runAs" -}}
+{{- if eq (include "defaultfalse" .Values.admin.securityContext.enabled) "true" }}
+runAsUser: {{ default 1000 .Values.admin.securityContext.runAsUser }}
+runAsGroup: 0
+  {{- if and (or (eq (include "defaulttrue" .Values.admin.initContainers.runInitDisk) "false") (eq (include "defaulttrue" .Values.admin.initContainers.runInitDiskAsRoot) "false")) (ne (toString (default 1000 .Values.admin.securityContext.runAsUser)) "0")  }}
+runAsNonRoot: true
```

Comment:
Does this actually have to be specified? It seems totally redundant, but I guess if the security rules require it then we have to specify it.

Reply:
This is needed to comply with [Require runAsNonRoot policy](https://kyverno.io/policies/pod-security/restricted/require-run-as-nonroot/require-run-as-nonroot/). Kubernetes does some extra checks if this is set to true.